src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -558,5 +562,16 @@ public void onAddAutoChannel() {
     preferencesService.storeInBackground();
     channelTextField.clear();
   }
+
+  public void onDebugLogToggle() {
+    Level logLevel = Level.DEBUG;
+    final LoggerContext loggerContext = ((Logger) LoggerFactory.getLogger(MethodHandles.lookup().lookupClass())).getLoggerContext();
+    loggerContext.getLoggerList()
+        .stream()
+        .filter(logger -> logger.getName().startsWith("com.faforever"))
+        .forEach(logger -> ((Logger) LoggerFactory.getLogger(logger.getName())).setLevel(logLevel));
+
+    log.info("Switching FA Forever logging configuration to {}", logLevel.levelStr);
+  }
```

Comment:
Business logic should be in a service

Reply:
OK, I've moved this logic to the PreferencesService, which I think is the most suitable Service for logging configurations.